From bcacab531f57b701053629b72b96b1ecf1e2336d Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Sun, 2 May 2021 17:19:11 +0200
Subject: [PATCH] [mobile] navNextFeature: fix skipping > 1 feature

Fix the condition in the while loop, so that it can actually continue
with the "continue" keyword. Without this patch, having the sshd
feature disabled would result in the fsType feature getting displayed,
even if it was disabled.
---
 modules/mobile/mobile.qml | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/modules/mobile/mobile.qml b/modules/mobile/mobile.qml
index 3397263..0343884 100644
--- a/modules/mobile/mobile.qml
+++ b/modules/mobile/mobile.qml
@@ -208,7 +208,9 @@ Page
                 id += 1;
                 continue;
             }
-        } while(false);
+
+            break;
+        } while (id < features.length);
 
         console.log("Navigating to feature: " + features[id]["name"]);
         return navTo(features[id]["screens"][0]);
-- 
2.26.3

